plugins/goodreads.koplugin/main.lua

```diff
@@ -0,0 +1,226 @@
+                        })
+                    end
+                end,
+            },
+        },
+    })
+end
+
+function GoodReads:updateSettings()
+    local hint_top
+    local text_top
+    local hint_bottom
+    local text_bottom
+    local text_info = "How to generate key and secret key:\n"..
+    "1. Go to https://www.goodreads.com/user/sign_up and create account\n" ..
+    "2. Register development key on page: https://www.goodreads.com/user/sign_in?rd=true\n" ..
+    "3. Your key and secret key are on https://www.goodreads.com/api/keys\n" ..
+    "4. Enter your generated key and secret key in settings (Login to GoodReads window)"
+    if self.goodreadersKey == "" then
+        hint_top = _("GoodReaders Key Not Set")
+        text_top = ""
+    else
+        hint_top = ""
+        text_top = self.goodreadersKey
+    end
+
+    if self.goodreadersSecret == "" then
+        hint_bottom = _("GoodReaders Secret Key Not Set (optional)")
+        text_bottom = ""
+    else
+        hint_bottom = ""
+        text_bottom = self.goodreadersKey
+    end
+    self.settings_dialog = MultiInputDialog:new {
+        title = _("Login to GoodReads"),
+        fields = {
+            {
+                text = text_top,
+                input_type = "string",
+                hint = hint_top ,
+            },
+            {
+                text = text_bottom,
+                input_type = "string",
+                hint = hint_bottom,
+            },
+        },
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    callback = function()
+                        self.settings_dialog:onClose()
+                        UIManager:close(self.settings_dialog)
+                    end
+                },
+                {
+                    text = _("Info"),
+                    callback = function()
+                        UIManager:show(InfoMessage:new{text = text_info })
+                    end
+                },
+                {
+                    text = _("Apply"),
+                    callback = function()
+                        self:saveSettings(MultiInputDialog:getFields())
+                        self.settings_dialog:onClose()
+                        UIManager:close(self.settings_dialog)
+                    end
+                },
+            },
+        },
+        width = Screen:getWidth() * 0.95,
+        height = Screen:getHeight() * 0.2,
+        input_type = "text",
+    }
+    self.settings_dialog:onShowKeyboard()
+    UIManager:show(self.settings_dialog)
+end
+
+function GoodReads:saveSettings(fields)
+    if fields then
+        self.goodreadersKey = fields[1]
+        self.goodreadersSecret = fields[2]
+    end
+    local settings = {
+        key = self.goodreadersKey,
+        secret = self.goodreadersSecret,
+    }
+    DEBUG(settings)
+    G_reader_settings:saveSetting("goodreads", settings)
+    DEBUG(self.goodreadersKey)
```

Comment:
Are these debug call still needed? If so, please prefix it with a descriptive text, something along the line of: `DEBUG('goodreaders key:', self.goodreadersKey)`